plutus-ledger-api/src/PlutusLedgerApi/V3/EvaluationContext.hs

```diff
@@ -32,6 +32,10 @@ a protocol update with the updated cost model parameters.
 mkEvaluationContext :: (MonadError CostModelApplyError m, MonadWriter [CostModelApplyWarn] m)
                     => [Integer] -- ^ the (updated) cost model parameters of the protocol
                     -> m EvaluationContext
-mkEvaluationContext = tagWithParamNames @V3.ParamName
-                    >=> pure . toCostModelParams
-                    >=> mkDynEvaluationContext Plutus.DefaultFunSemanticsVariant2
+mkEvaluationContext =
+    tagWithParamNames @V3.ParamName
+    >=> pure . toCostModelParams
+    >=> mkDynEvaluationContext
+        PlutusV3
+        [DefaultFunSemanticsVariant2]
+        (const Plutus.DefaultFunSemanticsVariant2)
```

Comment:
> Let me check: V1 and V2 have `if pv < conwayPV then ... else ...` here,  but it's safe to have `const` for V3 because we know that `PlutusV3` is going to be introduced at the same time as `conwayPV`, so a PlutusV3 script can never be executed with a lower PV, so there's no possiblity that `pv < conwayPV` here (but conceivably we might need `if pv < einsteinPV` or something in the future).  Is that right? 

I believe so, yes.

> Maybe we should try to explain that somewhere, since there's a LL/PV dependency that I don't think is captured in the code anywhere. Maybe we should have note containing a table like we do in the PLC specification (or a big table that contains all of the relevant information, since there are multiple tables in the spec).

Good point, I'll add a small table somewhere, but I'm not totally sure where given that the logic is spread across multiple files. I guess `PlutusLedgerApi.Common.Versions` indeed.